File: IrrgartenP4/irrgartenRuby/irrgarten/test_p2.rb
# test_p2.rb
require_relative 'monster'
require_relative 'player'
require_relative 'labyrinth'
require_relative 'game'
require_relative 'weapon'
require_relative 'shield'
require_relative 'dice'
require_relative 'game_state'

puts "===== Monster Test ====="
monster = Monster.new(5.0, 7.0)
monster.set_pos(2, 3)
puts "Monster Initial: #{monster}"
puts "Monster Attack: #{monster.attack}"
monster.got_wounded
puts "Monster after Wound: #{monster}"
puts "Monster is Dead: #{monster.dead?}"

puts "\n===== Player Test ====="
player = Player.new(1, 6.0, 8.0)
player.set_pos(1, 2)
puts "Player Initial: #{player}"
puts "Player Attack: #{player.attack}"
player.got_wounded
puts "Player after Wound: #{player}"
player.reset_hits
player.inc_consecutive_hits
puts "Player Defensive Energy: #{player.defensive_energy}"
puts "Player is Dead: #{player.dead?}"

puts "\n===== Labyrinth Test ====="
labyrinth = Labyrinth.new(5, 5)
puts "Initial Labyrinth:\n#{labyrinth}"
exit_pos = [4, 4]
labyrinth.set_exit(exit_pos[0], exit_pos[1])
puts "Labyrinth after setting exit:\n#{labyrinth}"

monster_pos = [2, 2]
labyrinth.add_monster(monster_pos[0], monster_pos[1], monster)
puts "Labyrinth after adding Monster:\n#{labyrinth}"

puts "Is Exit Position (4,4): #{labyrinth.exit_pos?(4, 4)}"
puts "Is Monster Position (2,2): #{labyrinth.monster_pos?(2, 2)}"
puts "Can Step on (3,3): #{labyrinth.can_step_on?(3, 3)}"
puts "Random Empty Position: #{labyrinth.random_empty_pos}"

puts "\n===== Game Test ====="
game = Game.new(2, 5, 5)
puts "Game Initial State: #{game.get_game_state}"

# Test player switching
game.next_player
puts "Game State after changing player: #{game.get_game_state}"

# Test logging
game.log_player_won
puts "Game Log after player wins:\n#{game.get_game_state[:log]}"

game.log_monster_won
puts "Game Log after monster wins:\n#{game.get_game_state[:log]}"

File: IrrgartenP5/IrrgartenJava/src/irrgarten/TestP2.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;


/**
 *
 * @author moham
 */// TestP2.java

public class TestP2 {
    public static void main(String[] args) {
        // Prueba de la clase Monster
        System.out.println("===== Monster Test =====");
        Monster monster = new Monster("i",5.0f, 7.0f);
        monster.setPos(2, 3);
        System.out.println("Monster Initial: " + monster);
        System.out.println("Monster Attack: " + monster.attack());
        monster.gotWounded();
        System.out.println("Monster after Wound: " + monster);
        System.out.println("Monster is Dead: " + monster.dead());

        // Prueba de la clase Player
        System.out.println("\n===== Player Test =====");
        Player player = new Player(1, 6.0f, 8.0f);
        player.setPos(1, 2);
        System.out.println("Player Initial: " + player);
        System.out.println("Player Attack: " + player.attack());
        player.gotWounded();
        System.out.println("Player after Wound: " + player);
        player.resetHits();
        player.incConsecutiveHits();
        System.out.println("Player Defensive Energy: " + player.defensiveEnergy());
        System.out.println("Player is Dead: " + player.dead());

        // Prueba de la clase Labyrinth
        System.out.println("\n===== Labyrinth Test =====");
        Labyrinth labyrinth = new Labyrinth(5, 5);
        System.out.println("Initial Labyrinth:\n" + labyrinth);
        int[] exitPos = {4, 4};
        labyrinth.setExit(exitPos[0], exitPos[1]);
        System.out.println("Labyrinth after setting exit:\n" + labyrinth);

        int[] monsterPos = {2, 2};
        labyrinth.addMonster(monsterPos[0], monsterPos[1], monster);
        System.out.println("Labyrinth after adding Monster:\n" + labyrinth);

        System.out.println("Is Exit Position (4,4): " + labyrinth.exitPos(4, 4));
        System.out.println("Is Monster Position (2,2): " + labyrinth.monsterPos(2, 2));
        System.out.println("Can Step on (3,3): " + labyrinth.canStepOn(3, 3));
        System.out.println("Random Empty Position: " + java.util.Arrays.toString(labyrinth.randomEmptyPos()));

        // Prueba de la clase Game
        System.out.println("\n===== Game Test =====");
        Game game = new Game(2, 5, 5);
        System.out.println("Game Initial State: " + game.getGameState());

        // Probar cambio de jugador
        game.nextPlayer();
        System.out.println("Game State after changing player: " + game.getGameState());

        // Prueba de logging
        game.logPlayerWon();
        System.out.println("Game Log after player wins:\n" + game.getGameState().getLog());

        game.logMonsterWon();
        System.out.println("Game Log after monster wins:\n" + game.getGameState().getLog());
    }
}
